src/app/shared/components/address-table/address-table.component.ts

```diff
@@ -77,16 +78,31 @@ export class AddressTableComponent implements OnInit {
       this.changeWithDelivery();
     }
     this.checkoutService.addresses$.subscribe(res => {
-      if (res && res.length && this.withDelivery) {
-        if (this.showAddresses.length === res.length - 1) {
-          this.selectedCustomerAddress = res.length - 1;
-        } else if (res.length === 1) {
-          this.selectedCustomerAddress = 0;
+      this.addresses = res;
+      if (this.withDelivery) {
+        if (this.deliveryDays === 3) { // should change days if added address is not tehran
+          if (this.province !== 'تهران') {
+            const duration_5 = this.durations.filter(el => el.delivery_days === 5)[0];
+            this.changeDurationType(duration_5._id, duration_5.delivery_days);
+            this.selectedCustomerAddress = this.addresses.length - 1;
+          } else {
+            this.showAddresses = this.addresses;
+            if (this.addresses && this.addresses.length && this.deliveryDays && (this.deliveryDays === 3)) {
+              this.tehranAddresses = this.addresses.filter(el => el.province === 'تهران');
```

Comment:
`this.tehranAddresses`
Please remove this variable (convert it to a function?) and calculate on the fly.